### ISSUE TITLE
Only download marked sponsorblock section. (call it "--sponsorblock-keep"?)

### ISSUE THREAD

**Issue by matthewrhodes** [yt-dlp/yt-dlp#8342]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a site-specific feature
- [X] I've verified that I'm running yt-dlp version **2023.10.07** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Example URLs

https://www.youtube.com/watch?v=a07pM4C3gaI

Only 0:48.895 to 1:09.181 would be downloaded. The output video from yt-dlp would be 21 seconds long.

### Provide a description that is worded well enough to be understood

Basically the inversion of what --sponsorblock-remove can do now, instead of cutting out sponsors and keeping the rest, it only downloads sponsors and the rest of the video is removed (or other marked section type).

Ideally, it shouldn't download the entire video, to save bandwidth.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
N/A
```
